docs/linters.md

```diff
@@ -477,6 +478,24 @@ linterConfig:
         message: prefer 'colour' over 'color' when referring to colours in field names
 ```
 
+## NumericBounds
+
+The `numericbounds` linter checks that numeric fields (`int32`, `int64`, `float32`, `float64`) have appropriate bounds validation markers.
+
+According to Kubernetes API conventions, numeric fields should have bounds checking to prevent values that are too small, negative (when not intended), or too large.
+
+This linter ensures that:
+- Numeric fields have both `+kubebuilder:validation:Minimum` and `+kubebuilder:validation:Maximum` markers
+- K8s declarative validation markers (`+k8s:minimum` and `+k8s:maximum`) are also supported
```

Comment:
I would swap the way we talk about these as we expect the `k8s` versions to take precedence in the long term